### PR TITLE
Allow setting config path via an env var

### PR DESCRIPTION
**Pull request by jedrichards**
Refs https://github.com/istarkov/babel-plugin-webpack-loaders/issues/41

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -179,7 +179,11 @@ export default function ({ types: t }) {
           return;
         }
 
-        const config = localInteropRequire(resolve(process.cwd(), configPath));
+        const config = localInteropRequire(resolve(
+          process.cwd(),
```
Comment (istarkov) [inline src/plugin.js]: I think if BABEL_PLUGIN_WEBPACK_LOADERS_CONFIG is set, there is no need in `process.cwd()` here.

Also it's really long long name ;-) 'BABEL_PLUGIN_WEBPACK_LOADERS_CONFIG'

IMHO it's better to allow just pass variables into config string, (it's easy to resolve such string with lodash template for example)

Comment (istarkov) [inline src/plugin.js]: Something like this
```javascript
var compiled = _.template(configPath);
const res =  compiled(process.env);
const config  = res === configPath
   ? localInteropRequire(resolve(process.cwd(), res))
   :  localInteropRequire(resolve(res))
```

Comment (jedrichards) [inline src/plugin.js]: That looks much better! WIll change ...

**Comment by jedrichards** [comment]
@istarkov How's it looking?

File: src/plugin.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { resolve, dirname, relative } from 'path';
 import { ResolverFactory, SyncNodeJsInputFileSystem } from 'enhanced-resolve';
 import { parse } from 'babylon';
+import template from 'lodash.template';
```
Comment (istarkov) [inline src/plugin.js]: IMHO it's better to use `import template from 'lodash/template'` as modern lodash already modular and I see no need in distinct libraries like 'lodash.template' now.

File: .babelrc [hunk 3]
```diff
@@ -27,6 +27,17 @@
           }
         ]
       ]
+    },
+    "DYNAMIC_CONFIG_TEST": {
+      "plugins": [
+        [
+          "./lib/plugin.js",
+          {
+            "config": "<%= CONFIG %>",
```
Comment (istarkov) [inline .babelrc]: Also I think here and in documentation is better to use `${CONFIG}` syntax as it like as javascript template literals. (This `<%= CONFIG%>` syntax is something from grunt ;-) path templates)

**Comment by istarkov** [comment]
Except two notes above all other is super!!!

File: src/plugin.js (hunk 2 above)
Comment (jedrichards) [inline src/plugin.js]: I don't use lodash much and was a bit confused about which was the modern form, and which was the old fashioned form! So I guess we use `lodash/template` these days to take advantage of npm3 and tree shaking etc.?

Will change it ...

Comment (istarkov) [inline src/plugin.js]: In this form we get advantage right now without tree shaking, with tree shaking as I understand we could just write
`import { template } from 'lodash'` and just `template` module will be imported. (But I haven't use tree shaking so may mistake)

File: .babelrc (hunk 3 above)
Comment (jedrichards) [inline .babelrc]: :+1:

File: src/plugin.js (hunk 2 above)
Comment (jedrichards) [inline src/plugin.js]: Ah makes sense, yes. Thanks for the info ...

**Comment by istarkov** [comment]
So good! Thank you!!! 
Now I can try ava ;-)

PS: One thing I can't get, why and how [appveyour](https://ci.appveyor.com/project/istarkov/babel-plugin-webpack-loaders/build/53) test was passed ;-) 
(_looks like new test was skipped at appveyour_)

**Comment by istarkov** [comment]
In npm now as 0.3.8

**Comment by jedrichards** [comment]
Nice, great!

Huh ... not sure what's going on with appveyor. Travis did all the tests. What was supposed to happen?

**Comment by istarkov** [comment]
IMHO windows should break on `CONFIG=$(pwd)/...` but I'm not sure ;-)

**Comment by jedrichards** [comment]
I can see that my test would be started as part of `npm test` on Travis, but where in appveyor.yml would start my tests?

**Comment by istarkov** [comment]
I've got it ;-)
https://github.com/istarkov/babel-plugin-webpack-loaders/blob/master/appveyor.yml#L24

appveyour runs tests not from `npm` ;-)

**Comment by jedrichards** [comment]
Ah sorry! I haven't used appveyor before ... assumed both happened from `npm test` :grimacing:

**Pull request merged**